storage/watchable_store_bench_test.go

```diff
@@ -0,0 +1,66 @@
+// Copyright 2015 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package storage
+
+import (
+	"fmt"
+	"math/rand"
+	"testing"
+)
+
+func BenchmarkWatchableStoreUnsyncedCancel(b *testing.B) {
+	s := newWatchableStore(tmpPath)
+	defer cleanup(s, tmpPath)
+
+	b.ReportAllocs()
+
+	watcherSize := b.N
+	if watcherSize < 10 {
+		return
+	}
+
+	b.StopTimer()
+
+	keys := [][]byte{}
+	for j := 0; j < watcherSize; j++ {
+		k := []byte(fmt.Sprint("foo", j))
+		s.Put(k, []byte("bar"))
+		keys = append(keys, k)
+	}
```

Comment:
could we watch on the same key? From the perspective of implementation, it has no difference.